packages/css/src/rules/S7923/rule.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * SonarQube JavaScript Plugin
+ * Copyright (C) 2011-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the Sonar Source-Available License Version 1, as published by SonarSource SA.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the Sonar Source-Available License for more details.
+ *
+ * You should have received a copy of the Sonar Source-Available License
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
+// https://sonarsource.github.io/rspec/#/rspec/S7923/css
+import stylelint, { type PostcssResult } from 'stylelint';
+import type PostCSS from 'postcss';
+import postcssValueParser from 'postcss-value-parser';
+import unitConverter, { type CSSUnits } from 'css-unit-converter';
+
+const ruleName = 'sonar/no-restrict-orientation';
+const angleRegex = /(?<angle>-?\d+(?:\.\d+)?)\s*(?<unit>deg|rad|grad|turn)/i;
+
+// exported for testing purpose
+export const messages = {
+  locked: 'Do not lock content to a specific display orientation.',
+};
+
+const incorrectAngles = [90, 270];
+const tolerance = 30;
+
+const ruleImpl: stylelint.RuleBase = () => {
+  return (root: PostCSS.Root, result: PostcssResult) => {
+    root.walkAtRules((atRule: PostCSS.AtRule) => {
+      if (atRule.name === 'media' && atRule.params.includes('orientation')) {
+        atRule.walkDecls((decl: PostCSS.Declaration) => {
+          postcssValueParser(decl.value).walk((node: postcssValueParser.Node) => {
+            if (isAngleInvalid(getAngle(node))) {
+              stylelint.utils.report({
+                ruleName,
+                result,
+                message: messages.locked,
+                node: decl,
+              });
+            }
+          });
+        });
+      }
+    });
+  };
+};
+
+function isAngleInvalid(angle: number | undefined) {
+  if (angle === undefined) {
+    return false;
+  }
+  // Check if angle is within range of 90° (60° to 120°) or within range of 270° (240° to 300°)
+  return incorrectAngles.some(incorrectAngle => {
+    const lowerBound = incorrectAngle - tolerance;
+    const upperBound = incorrectAngle + tolerance;
+    return angle >= lowerBound && angle <= upperBound;
```

Comment:
Also, not sure if you want strict or not script to allow for the 60, 120 and 240 and 300
```suggestion
    return Math.abs(angle - incorrectAngle) <= tolerance
```